scripts/lint.sh

```diff
@@ -98,6 +98,15 @@ function test_require_equal_len {
   fi
 }
 
+function test_require_nil {
+  if grep -R -zo -P 'require.+?nil\)\n' .; then
```

Comment:
Is there a reason why we only check for `nil` at the end as a last param rather than anywhere inside a require?

Reply:
It would falsely flag stuff we do use like:
```
require.Equal((*txs.Tx)(nil), mpool.Get(tx.ID()))
```

Reply:
I think what we're really trying to do is this:
```suggestion
  if grep -R -o -P 'require\..+?(!|=)= nil' .; then
```